### PR TITLE
Clone urlsearchparams to avoid mutation

### PR DESCRIPTION
**Pull request by jimmywarting**
...and set correct content-type in response constructor

followup on #520

### CODE REVIEW

**Comment by bitinn** [comment]
@jimmywarting I think this is now fine, could you fix up the indent for match with current format? Also double check the coverage for `extractContentType` is still 100% now that you modified it slightly.

(use `npm run report`)

**Comment by jimmywarting** [comment]
extractContentType seems fine.

report:
```
----------------|----------|----------|----------|----------|-------------------|
File            |  % Stmts | % Branch |  % Funcs |  % Lines | Uncovered Line #s |
----------------|----------|----------|----------|----------|-------------------|
All files       |    95.41 |    92.42 |      100 |    95.35 |                   |
 body.js        |     88.4 |    82.24 |      100 |    88.07 |... 33,463,508,509 |
 fetch-error.js |      100 |      100 |      100 |      100 |                   |
 headers.js     |      100 |      100 |      100 |      100 |                   |
 index.js       |      100 |      100 |      100 |      100 |                   |
 request.js     |      100 |      100 |      100 |      100 |                   |
 response.js    |      100 |      100 |      100 |      100 |                   |
----------------|----------|----------|----------|----------|-------------------|
```

File: src/response.js [hunk 1]
```diff
@@ -23,12 +23,20 @@ export default class Response {
 		Body.call(this, body, opts);
 
 		const status = opts.status || 200;
+    const headers = new Headers(opts.headers)
```
Comment (bitinn) [inline src/response.js]: you probably missed this line.

**Comment by bitinn** [comment]
Alright I have merged it, but I suspect `res.body` will change in future as it's not spec compliant to leave it as a buffer (same goes for other type of body input I guess, basically #210)

https://developer.mozilla.org/en-US/docs/Web/API/Body/body

**Comment by jimmywarting** [comment]
> EDIT: actually, it might be ok to leave them as a buffer internally, as long as res.body getter is guaranteed to be a stream

That was my idea too. I think I will solve #210 by making two PR, one where i unify everything to a buffer to remove redundant code. and one after that would make it easier to create a stream when all the internal is either a buffer or a stream

**Comment by bitinn** [comment]
@jimmywarting sounds good

**Pull request merged**